### PR TITLE
Use Enso font

### LINKED ISSUE #8485: Use Enso Font in new GUI
As in title. The fonts are defined in https://github.com/enso-org/font repository. They should be downloaded during `npm install` and ready to use in new GUI codebase. For details, see https://github.com/enso-org/enso/issues/7197.

### PR DESCRIPTION
**Pull request by somebody1234**
### Pull Request Description
- Closes #8485

### Important Notes
None

### Screenshots
Before:
image
image
After:
image
image

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [x] ~~The documentation has been updated, if necessary.~~
- [x] Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.
- [x] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- All code has been tested:
  - [x] ~~Unit tests have been written where possible.~~
  - [ ] If GUI codebase was changed, the GUI was tested when built using `./run ide build`.

### CODE REVIEW

**Comment by somebody1234** [comment]
adding @kazcw as reviewer as he implemented the corresponding PR for GUI1.
(is there anywhere else that needs this font? most places should have it due to CSS cascading, but it may be conceptually clearer to explicitly add it in more places. alternatively, maybe it's currently being added in too many places - e.g. both `main.css` and `App.vue`)

File: app/gui2/stories/story.css [hunk 1]
```diff
@@ -1,6 +1,13 @@
+@import url('../src/assets/font-enso-code.css');
+
+:root {
+  --font-code: 'Enso Code', system-ui, -apple-system, BlinkMacSystemFont, 'Segoe UI', 'Roboto',
```
Comment (Frizi) [inline app/gui2/stories/story.css]: Why is it necessary to repeat this css here? Shouldn't the declaration from `base.css` already work here?

**Review by Frizi — APPROVED** [review]

File: app/gui2/stories/story.css (hunk 1 above)
Comment (somebody1234) [reply app/gui2/stories/story.css]: do we import `base.css` in Histoire though?

Comment (somebody1234) [reply app/gui2/stories/story.css]: I guess I'll test this, and remove if it is unnecessary

**Review by kazcw — COMMENTED** [review]
One thing to watch out for is that the Enso Font is not intended to be used for non-code text without performing some glyph substitutions first (affecting period/space/hyphen): https://github.com/enso-org/font#using-enso-font-in-regular-sentences

Also, some visualizations will break if a non-monospaced font (like Enso Font) is used.

**Comment by somebody1234** [comment]
hm... should we even be using Enso Font instead of M PLUS 1 in those cases then?

**Comment by somebody1234** [comment]
@kazcw also, which visualizations will break btw?

**Comment by kazcw** [comment]
> hm... should we even be using Enso Font instead of M PLUS 1 in those cases then?

Indeed--in GUI1 we were using three fonts:
- Enso Font for code
-  M PLUS 1 for most non-code
- Some monospace font (DejaVu Sans Mono?) for the text grid visualization (it looks like that isn't changed here)

It's a little inefficient since Enso Font and M PLUS 1 are mostly the same. Some day as an optimization (rather than glyph substitutions) we could reduce the Enso Font to only include a few glyphs, and specify the glyph fallbacks in CSS so that our code font is Enso Font overlayed on the base M PLUS 1 font.

**Comment by somebody1234** [comment]
hmm... i think i'll switch back to M PLUS 1 for regular text, i can't imagine it being very performant to do text replacement on every single text span in Vue

**Pull request merged**